public/components/datasources/components/new/configure_s3_datasource.tsx

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  EuiPanel,
+  EuiTitle,
+  EuiSpacer,
+  EuiText,
+  EuiLink,
+  EuiFormRow,
+  EuiFieldText,
+  EuiTextArea,
+  EuiButton,
+} from '@elastic/eui';
+import React, { useState } from 'react';
+import { OPENSEARCH_DOCUMENTATION_URL } from '../../../../../common/constants/data_connections';
+import { QueryPermissionsConfiguration } from '../manage/query_permissions';
+
+interface ConfigureS3DatasourceProps {
+  roles: Array<{ label: string }>;
+  selectedQueryPermissionRoles: Array<{ label: string }>;
+  setSelectedQueryPermissionRoles: React.Dispatch<React.SetStateAction<Array<{ label: string }>>>;
```

Comment:
minor, the `Array<{ label: string }>` keeps appearing, maybe create a type for it? and why not just `roles: EuiComboBoxOptionOption[]` or maybe
```typescript
type Role = EuiComboBoxOptionOption
roles: Role[]
```

Reply:
Good suggestion - [680763b](https://github.com/opensearch-project/dashboards-observability/pull/1049/commits/680763b566a24a70d93f79214331182f495ed291), thanks!